plugins/faucet/plugin.go

```diff
@@ -150,21 +122,13 @@ func run(plugin *node.Plugin) {
 		}
 		plugin.LogInfo("Waiting for node to have sufficient access mana... done")
 
-		plugin.LogInfof("Deriving faucet state from the ledger...")
-
-		// determine state, prepare more outputs if needed
-		if err := _faucet.DeriveStateFromTangle(ctx); err != nil {
-			plugin.LogErrorf("failed to derive state: %s", err)
-			return
-		}
-		plugin.LogInfo("Deriving faucet state from the ledger... done")
-
-		defer fundingWorkerPool.Stop()
-		defer preparingWorkerPool.Stop()
-
+		_faucet.DeriveStateFromTangle()
 		initDone.Store(true)
 
+		go _faucet.Start(ctx, requestChan)
```

Comment:
Is a new goroutine necessary here? In the next line we block on the ctx channel, and the goroutine waits for the same channel. Maybe it's not necessary to wait for the context here but simlpy make the `_faucet.Start` blocking?